repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/core/manifest/VersionAddedManifestTest.groovy

```diff
@@ -0,0 +1,176 @@
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.core.manifest
+
+import framework.ReposeValveTest
+import org.apache.commons.io.FileUtils
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.PortFinder
+
+import java.util.concurrent.TimeoutException
+import java.util.jar.Attributes
+import java.util.jar.JarFile
+import java.util.jar.Manifest
+
+/**
+ * Created by jennyvo on 4/12/16.
+ */
+class VersionAddedManifestTest extends ReposeValveTest {
+
+    def setupSpec() {
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(properties.targetPort)
+    }
+
+    def cleanupSpec() {
+        deproxy.shutdown()
+    }
+
+    static def params
+
+    def "Verify Version added to manifest"() {
+        setup: "starts repose with installation configs"
+        def params = properties.getDefaultTemplateParams()
+        def nextPort = PortFinder.Singleton.getNextOpenPort()
+
+        //note: Order matters here. The common directory overwrites some of the configs from the core directory.
+        //      This means that the core configs we provide may not get tested, but due to the structure of our tests,
+        //      this is currently "hard" to fix.
+        repose.configurationProvider.applyConfigs("../../../../installation/configs/core", params)
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("../../../../installation/configs/extensions", params)
+        repose.configurationProvider.applyConfigs("../../../../installation/configs/filters", params)
+        String systemModelTemp = "${repose.configurationProvider.reposeConfigDir}/system-model.cfg.xml.${nextPort}"
+        String systemModelSource = "${repose.configurationProvider.reposeConfigDir}/system-model.cfg.xml"
+        new File(systemModelTemp).withWriter {
+            out ->
+                new File(systemModelSource).eachLine {
+                    line ->
+                        out << line.replaceAll("http-port=\"8080\"", "http-port=\"${nextPort}\"")
+                }
+        }
+        FileUtils.copyFile(new File(systemModelTemp), new File(systemModelSource))
+
+        repose.start()
```

Comment:
Is there a reason we need to start Repose to inspect the manifests of our artifacts/